src/components/molecules/Visualizer/Widget/index.tsx

```diff
@@ -21,6 +23,12 @@ export type Props<PP = any, SP = any> = {
   sceneProperty?: SP;
   pluginProperty?: PP;
   pluginBaseUrl?: string;
+  pluginModalContainer?: HTMLElement | DocumentFragment;
+  shownPluginModalInfo?: PluginModalInfo;
+  showPluginModal?: (modalInfo?: PluginModalInfo) => void;
+  pluginPopupContainer?: HTMLElement | DocumentFragment;
+  shownPluginPopupInfo?: PluginPopupInfo;
+  showPluginPopup?: (popupInfo?: PluginPopupInfo) => void;
```

Comment:
There are many fields, so why not using the external type?

`Widget/index.tsx`

```ts
import Plugin, { type Props as PluginProps } from "../Plugin";

export type Props = {
  // ...
} & PluginProps;
```

Also you can consider defining common props in Plugin:

`Plugin/index.tsx`

```
export type CommonProps = {
  // add other fields
  pluginModalContainer?: HTMLElement | DocumentFragment;
  shownPluginModalInfo?: PluginModalInfo;
  showPluginModal?: (modalInfo?: PluginModalInfo) => void;
  pluginPopupContainer?: HTMLElement | DocumentFragment;
  shownPluginPopupInfo?: PluginPopupInfo;
  showPluginPopup?: (popupInfo?: PluginPopupInfo) => void;
};

export type Props = {
	// not used props externally (e.g. not used props at Visualzier/index.tsx)
} & CommonProps;
```

Then `Widget/index.tsx`

```ts
import Plugin, { type CommonProps as PluginProps } from "../Plugin";

export type Props = {
  // ...
} & PluginProps;
```

Reply:
Updated.